src/merge_tools.rs

```diff
@@ -416,13 +424,48 @@ pub fn edit_diff(
     Ok(right_tree_state.current_tree_id().clone())
 }
 
+/// Generates textual diff by the specified `tool`, and writes into `writer`.
+pub fn generate_diff(
+    writer: &mut dyn Write,
+    left_tree: &Tree,
+    right_tree: &Tree,
+    matcher: &dyn Matcher,
+    tool: &MergeTool,
+) -> Result<(), DiffGenerateError> {
+    let store = left_tree.store();
+    let diff_wc = check_out_trees(store, left_tree, right_tree, matcher)?;
+    // TODO: Add support for tools without directory diff functionality?
+    // TODO: Somehow propagate --color to the external command?
+    let patterns = diff_wc.to_command_variables();
+    let mut cmd = Command::new(&tool.program);
+    cmd.args(interpolate_variables(&tool.diff_args, &patterns));
+    tracing::info!(?cmd, "Invoking the external diff generator:");
+    let mut child = cmd
+        .stdin(Stdio::null())
+        .stdout(Stdio::piped())
+        .spawn()
+        .map_err(|source| ExternalToolError::FailedToExecute {
+            tool_binary: tool.program.clone(),
+            source,
+        })?;
+    io::copy(&mut child.stdout.take().unwrap(), writer).map_err(ExternalToolError::Io)?;
+    let exit_status = child.wait().map_err(ExternalToolError::Io)?;
```

Comment:
Perhaps, we shouldn't check the exit code since the code may represent whether the contents are different or not.